### ISSUE TITLE
Infinite loop with evalcache

### ISSUE THREAD

**Issue by bduff9** [nvbn/thefuck#1178]
**The output of `thefuck --version` (something like `The Fuck 3.1 using Python
3.5.0 and Bash 4.4.12(1)-release`):**

    ❯ thefuck --version
    The Fuck 3.30 using Python 3.9.2 and ZSH 5.8

**Your system (Debian 7, ArchLinux, Windows, etc.):**

    macOS 11.2.3

**How to reproduce the bug:**

Trying to speed up my ZSH loading time (was ~7secs, currently down to just under 2secs) and ran across the evalcache plugin to help speed up eval calls.  The plugin is here: https://github.com/mroth/evalcache

Basically, you replace the eval call `eval "$(direnv hook zsh)"` with `_evalcache direnv hook zsh`.  I have done this for 4 other eval calls in my .zshrc.  However, when I convert `eval $(thefuck --alias --enable-experimental-instant-mode)` to `_evalcache thefuck --alias --enable-experimental-instant-mode`, I get an infinite loading shell that seems to execute my .zshrc file over and over until manually killed.

I tried this without the instant mode flag too, and no difference.

**The output of The Fuck with `THEFUCK_DEBUG=true` exported (typically execute `export THEFUCK_DEBUG=true` in your shell before The Fuck):**

I added this to my .zshrc file before running the eval, however no output was written during this infinite loop.

**If the bug only appears with a specific application, the output of that application and its version:**

N/A

**Anything else you think is relevant:**

Unsure

**Comment by scorphus** [comment]
What happens if you leave only `_evalcache thefuck --alias` and remove (or comment) the other `_evalcache` calls?

**Comment by bduff9** [comment]
@scorphus So interestingly enough, it seems like it's working now.  I had played around with this a bit before when troubleshooting and tried various combinations of evals, which is how I narrowed it down specifically to thefuck.  But now, uncommenting it allows it to work with my other 2 evalcache calls too.

The only thing I can offer is after debugging and then writing this github issue, I cleared out the evalcache file for thefuck.  Maybe the first time it ran it produced bad output of some kind which got cached, so clearing that file fixed it?

Either way, this seems resolved now :)
